common-lib/src/main/scala/com/gu/mediaservice/lib/json/JsonByteArrayUtil.scala

```diff
@@ -0,0 +1,55 @@
+package com.gu.mediaservice.lib.json
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream}
+import java.nio.charset.StandardCharsets
+import java.util.zip.{GZIPInputStream, GZIPOutputStream}
+
+import play.api.libs.json._
+
+import scala.io.Source.fromInputStream
+
+object JsonByteArrayUtil extends PlayJsonHelpers {
+  private val compressionMarkerByte: Byte = 0x00.toByte
+
+  private def compress(bytes: Array[Byte]): Array[Byte] = {
+    val outputStream = new ByteArrayOutputStream()
```

Comment:
I feel like I _should_ be a bit intimidated by this adding an extra byte stuff, but ... this is really clear 👍

Reply:
Heavily inspired by [Tag Manager](https://github.com/guardian/tagmanager/blob/4d78a1e5ab3c1588ea89116639cb39bc19443e0d/app/services/ThriftSerializer.scala) regarding using the first byte to indicate what kind of thing we have.